### PR TITLE
auth/service: use address option since router may not be configured

### PR DESCRIPTION
**Pull request by ben-toogood**
Auth is the first thing which gets setup during Initialisation, hence the router may not be configured. To combat this, we can now manually set the auth address.

### CODE REVIEW

File: auth/service/service.go [hunk 1]
```diff
@@ -30,10 +30,6 @@ func (s *svc) Init(opts ...auth.Option) {
 		o(&s.options)
 	}
 
-	if s.options.Client == nil {
```
Comment (ben-toogood) [inline auth/service/service.go:33]: This check is already done in NewAuth()

File: config/cmd/cmd.go [hunk 2]
```diff
@@ -570,6 +575,63 @@ func (c *cmd) Before(ctx *cli.Context) error {
 	microClient := wrapper.CacheClient(cacheFn, grpc.NewClient())
 	microClient = wrapper.AuthClient(authFn, microClient)
 
+	// Setup auth options
+	authOpts := []auth.Option{auth.WithClient(microClient)}
+	if len(ctx.String("auth_address")) > 0 {
+		authOpts = append(authOpts, auth.Addrs(ctx.String("auth_address")))
+	}
+	if len(ctx.String("auth_id")) > 0 || len(ctx.String("auth_secret")) > 0 {
```
Comment (domwong) [inline config/cmd/cmd.go:583]: Should this be an `&&` rather than an `||`?

File: config/cmd/cmd.go [hunk 3]
```diff
@@ -570,6 +575,63 @@ func (c *cmd) Before(ctx *cli.Context) error {
...
+		a, ok := c.opts.Auths[name]
+		if !ok {
+			logger.Fatalf("Unsupported auth: %s", name)
+		}
+		*c.opts.Auth = a(authOpts...)
+		serverOpts = append(serverOpts, server.Auth(*c.opts.Auth))
+	} else if len(authOpts) > 0 {
+		(*c.opts.Auth).Init(authOpts...)
+	}
+
```
Comment (domwong) [inline config/cmd/cmd.go:634]: Why did you move this block of code up the order?

File: config/cmd/cmd.go (hunk 2 above)
Comment (ben-toogood) [reply config/cmd/cmd.go:583]: No, if either value is set we want to set the option. If it's incorrect the auth service will be responsible for returning an error on startup. We only really have the check so we don't call Init without reason.

File: config/cmd/cmd.go (hunk 3 above)
Comment (ben-toogood) [reply config/cmd/cmd.go:634]: So that auth is configured before calls to the router / registry get made since these could be authorised.

**Review by domwong — APPROVED** [review]
